dask/array/core.py

```diff
@@ -891,20 +896,58 @@ def store(sources, targets, lock=True, regions=None, compute=True, **kwargs):
         except AttributeError:
             dsk = {}
 
-        update = insert_to_ooc(tgt, src, lock=lock, region=reg)
-        keys.extend(update)
-
-        update.update(dsk)
-        updates.update(update)
+        store_name = 'store-%s' % src.name
+        store_dsk = insert_to_ooc(src, tgt, lock=lock, region=reg, keep=keep)
+
+        if keep:
+            load_name = 'load-%s' % src.name
+            load_dsk = retrieve_from_ooc(store_dsk)
+
+            if compute:
+                from ..delayed import Delayed
+                for each_store_key in store_dsk:
+                    store_dsk[each_store_key] = (
+                        Delayed(
+                            each_store_key,
+                            dict(sharedict.merge(
+                                (
+                                    each_store_key,
+                                    {each_store_key: store_dsk[each_store_key]}
+                                ),
+                                dsk,
+                                src.dask
+                            ))
+                        ).persist().dask[each_store_key]
+                    )
```

Comment:
`.persist` is blocking when using the single machine scheduler.  We definitely don't want to call it in a loop.

FWIW I find this statement difficult to parse.  That's subjective though.

Reply:
Sure we can drop all of these into one `persist` call. Expect that will make it easier to parse as well.

Reply:
Should add that generally what we are trying to accomplish here is submitting the Dask Graph associated with storing each chunk.

Reply:
Also valuable to combine `persist` calls to reuse intermediates.

FWIW I think this statement should now be a bit cleaner and more performant.